Sage/carbon
# feat(anchor-navigation): describe anchor navigation using design tokens, update test - FE-4730

### Proposed behaviour

Describes anchor navigation element using the tokens of the design.
Describes focus state using outline:  `var(--borderWidth300) solid var(--colorsSemanticFocus500)`
Increases the width of the indicator bar to `4px`. 
Updates tests to include changes.

### Current behaviour

Anchor navigation component use the theme styled-component property.
Current focus state is described by a box-shadow:  box-shadow: `rgb(255 181 0) 0px 0px 6px`;
Current width of the indicator bar is `3px`. 

### Checklist




#### QA


### Additional context

![image](https://user-images.githubusercontent.com/44143630/150160419-a9109bee-e89d-4dbe-87f8-36353d5a7eef.png)
![image](https://user-images.githubusercontent.com/44143630/150160443-510bdd58-9da1-4f7f-87a0-d2cd67ee6b26.png)

### Testing instructions

The correct appearance can be checked in the component: Anchor Navigation.
For theme: mint and aegean there should be no differences.
Use devtools to check if anchor navigation use css variables.
